prompterator/main.py

```diff
@@ -411,6 +427,23 @@ def set_up_ui_generation():
         label_visibility="collapsed",
         key=c.PROMPT_COMMENT_KEY,
     )
+    selected_model: c.ModelProperties = st.session_state.model
+    available_structured_output_settings = selected_model.supports_structured_output
+
+    # Allow structured outputs only if the model allows other implementation
+    # than NONE, other implementations currently include FUNCTION_CALLING
+    # and RESPONSE_FORMAT. Models by default do not require this parameter to be set.
+    structured_output_available = (
+        len(set(available_structured_output_settings) - {c.StructuredOutputImplementation.NONE})
+        > 0
+    )
```

Comment:
@gkaretka let's simplify this. Your implementation is robust in the sense that it won't break if one adds further values into the `c.StructuredOutputImplementation` enum. But it is fragile in the sense that it's hard to understand and accompanied by a comment which will become out-dated if `StructuredOutputImplementation` changes.

```suggestion
    structured_output_available = c.StructuredOutputImplementation.FUNCTION_CALLING in available_structured_output_implementations or c.StructuredOutputImplementation.STRUCTURED_OUTPUT in available_structured_output_implementations
```

Reply:
The thing here is that when you add new `StructuredOutputImplementation` this stops working. I'm not huge fan of enumerating states that are in majority (eg. here, allowed states are in majority, and might grow in the future), on the other hand the only state when we don't want to enable the toggle is when there is `None` implementation available. What do you think?